rpc/wrtc_client_channel.go

```diff
@@ -336,29 +333,42 @@ func (ch *webrtcClientChannel) writeReset(stream *webrtcpb.Stream) error {
 
 // taken from
 // https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/logging/zap/client_interceptors.go
-
-func logFinalClientLine(ctx context.Context, startTime time.Time, err error, msg string) {
+func logFinalClientLine(logger utils.ZapCompatibleLogger, startTime time.Time, err error, msg string) {
 	code := grpc_logging.DefaultErrorToCode(err)
 	level := grpc_zap.DefaultCodeToLevel(code)
+
+	// this calculation is done because duration.Milliseconds() will return an integer, which is not precise enough.
+	duration := float32(time.Since(startTime).Nanoseconds()/1000) / 1000
+	fields := []any{}
 	if err == nil {
 		level = zap.DebugLevel
+	} else {
+		fields = append(fields, "error", err)
+	}
+	fields = append(fields, "grpc.code", code.String(), "grpc.time_ms", duration)
+
+	if logger.Level().Enabled(level) {
```

Comment:
Why is this necessary? I would expect if the level were not enabled, then the calls to `logger.Debuw`/`logger.Infow`/etc... would early return. Are we not observing that?

Reply:
ya this is probably unnecessary, let me check. I was thinking too much about replicating the functionality of grpc_zap.DefaultMessageProducer, which also does a check before logging

Reply:
Cool, definitely speak up if removing this is problematic.

If removing is not problematic, then we can cascade and delete the [interface change](https://github.com/viamrobotics/goutils/pull/375/files#diff-ff87b7c4777a35588053a509583d66c9f404ccbea9e1c71d2a5f224d7ad1323eR28) that added `Level` and the dummy implementations for the [testing](https://github.com/viamrobotics/goutils/pull/375/files#diff-55a2db70576eb079880167200221ea3787a01fa4bbc958503769eba24084d842R68) [cases](https://github.com/viamrobotics/goutils/pull/375/files#diff-55a2db70576eb079880167200221ea3787a01fa4bbc958503769eba24084d842R87).

Reply:
updated to remove